### PR TITLE
New article: TWVycnkgQ2hyaXN0bWFzIDop

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Hello,

Here is another little story about elves, Perl and MIME:base64 🎅🏼

### CODE REVIEW

**Comment by thibaultduponchelle** [comment]
@vanHoesel Would you be available to review this article? Please 🙏🏼

**Comment by thibaultduponchelle** [comment]
Hi @adamcrussell

As discussed, if you could help editing and correcting this article, it would be a great 👍🏼

**Review by adamcrussell — APPROVED** [review]
I like this! I especially like the style and backstory. I believe the Advent Calendar entries should have a distinct holiday theme like this. 

1. The spelling sirup is much less common than syrup. No need to change it, in my opinion.
2. I would add maybe a little more content. Another example or two. This could be fine as is, but if so inspired I think the typical Advent Calendar entry is a little longer.

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Title: TWVycnkgQ2hyaXN0bWFzIDop
+Topic: MIME::Base64
+
+=encoding utf8
+
+=head1 TWVycnkgQ2hyaXN0bWFzIDop
+
+=head2 Perl birthday party
+
+On 18th of December, to celebrate the Perl birthday, elves organized a big party with cakes, waffles, pancakes and maple sirup.
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:11]: > 
```suggestion
On the 18th of December, to celebrate Perl's birthday, the elves organized a big party with cakes, waffles, pancakes and maple sirup.
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+They party like crazy but only until exactly midnight where everyone went to bed.
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:13]:
```suggestion
They partied like crazy but only until exactly midnight when everyone went to bed.
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+The day after, elves walked to North Pole warehouse. To open the door and disable the alarm, as every day, they called the dedicated elf "spof" to enter the code.
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:15]:
```suggestion
The day after, the elves walked to North Pole warehouse. To open the door and disable the alarm, as every day, they called the dedicated elf "spof" to enter the code.
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+He typed the code and the security LCD printed "ERROR".
+
+- Type it again, said another elf
+
+He proceed and got once again "ERROR"
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:20]:
```suggestion
He proceeded and, once again, got "ERROR".
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+- WWVz
+
+=head2 Deciphering the code
+
+One of the elves has little experience with IT and he immediately thought he would test L<MIME::Base64> to translate this gibberish:
+
+=begin perl
+
+	use MIME::Base64;
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:40]:
```suggestion
	use MIME::Base64 qw( decode_base64 );
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+=head2 Secret is compromised!
+
+But wait, now everyone is aware of the secret code, including you! This is not acceptable!
+
+So some senior elves decided to change the code:
+
+=begin perl
+
+	use Env;
+	use MIME::Base64;
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:57]:
```suggestion
	use MIME::Base64 qw( encode_base64 );
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+	print encode_base64($ENV{SECRET});
+
+=end perl
+
+=head2 Conclusion
+
+You should never feed a L<Gremlin|https://en.wikipedia.org/wiki/Gremlins> after midnight but you also need to monitor closely the maple sirup consumption of elves!
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:65]:
```suggestion
You should never feed L<Gremlin|https://en.wikipedia.org/wiki/Gremlins> after midnight but you also need to monitor closely the maple sirup consumption of elves!
```

**Review by oalders — CHANGES_REQUESTED** [review]
Love it! Just a few minor things. 🧝🏻

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+- TWVycnkgQ2hyaXN0bWFzIDop
+
+- Are you sure?
+
+- WWVz
+
+=head2 Deciphering the code
+
+One of the elves has little experience with IT and he immediately thought he would test L<MIME::Base64> to translate this gibberish:
```
[the remark this answers is not in the document]
Comment (thibaultduponchelle) [reply 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:36]: Yes, good catch thx

**Comment by thibaultduponchelle** [comment]
I added a paragraph upon remark of @adamcrussell and fixed all requested changes. 

Except edits in the new paragraph, I think it could be ready to merge.

**Comment by thibaultduponchelle** [comment]
```
$ perl t/article_pod.t 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod 
# Subtest: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
    # Subtest: header for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
        ok 1 - Required field <title> is there
        ok 2 - Required field <topic> is there
        ok 3 - Required field <author> is there
        1..3
    ok 1 - header for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
    # Subtest: pod for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
        ok 1 - Pod checks passes
        1..1
    ok 2 - pod for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
    1..2
ok 1 - 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
1..1
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+=begin perl
+
+	use Env;
+	use MIME::Base64 qw( encode_base64 );
+
+	print encode_base64($ENV{SECRET});
+
+=end perl
+
+The above script is using environment variable to pass the secret.
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:63]:
```suggestion
The above script is using an environment variable to pass the secret.
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+=head2 Make perl speak Spof's dialect
+
+Aside mapple syrup, Spof enjoys doing some Perl scripting for himself the evening, so after this event he did (or at least he I<tried>).
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:67]:
```suggestion
Aside from maple syrup, Spof enjoys doing some Perl scripting for himself the evening, so after this event he did (or at least he I<tried>).
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+Being still confused he tried to write C<cHJpbnQgIk1lcnJ5IENocmlzdG1hcyA6KSIK> but it was not working.
+
+Confused but lazy, he dig the L<MetaCPAN|https://metacpan.org/> for a solution and went with L<Acme::Base64>:
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:71]:
```suggestion
Confused but lazy, searched the L<MetaCPAN|https://metacpan.org/> for a solution and went with L<Acme::Base64>:
```

File: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+=begin perl
+
+	use Acme::Base64;
+
+	cHJpbnQgIk1lcnJ5IENocmlzdG1hcyA6KSIK
+
+=end perl
+
+This way, C<perl> interpreter could understand the B<base64> dialect spoken by Spof! Magic!
```
Comment (oalders) [inline 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:81]:
```suggestion
This way, the C<perl> interpreter could understand the B<base64> dialect spoken by Spof! Magic!
```

**Review by oalders — CHANGES_REQUESTED** [review]
Great stuff! Just minor suggestions.

**Comment by thibaultduponchelle** [comment]
@oalders I updated, ready to merge?

**Comment by oalders** [comment]
@thibaultduponchelle, sure. Could you squash it down to a single commit?

**Comment by thibaultduponchelle** [comment]
Done

**Review by oalders — APPROVED** [review]
\o/

**Comment by oalders** [comment]
I guess we can also squash via the merge button. I had forgotten about that.

**Pull request merged**